packages/performance/src/services/perf_logger.ts

```diff
@@ -120,18 +120,12 @@ export function logTrace(trace: Trace): void {
     return;
   }
 
-  if (
-    !settingsService.loggingEnabled ||
-    !settingsService.logTraceAfterSampling
-  ) {
-    return;
-  }
-
   if (isPerfInitialized()) {
```

Comment:
Nit: We can always use the promise chain without checking `isPerfInitialized()`

Reply:
That is true, however, Fireperf initialization state will only change once and will stay true after initialization. Check for initialization state in synchronized way is more straightforward and efficient.